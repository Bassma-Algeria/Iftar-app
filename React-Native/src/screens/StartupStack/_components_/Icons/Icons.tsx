import React from 'react';
import {Image} from 'react-native';
import {ICONS} from '../../../../utils/constants/Icons.ts';
interface Props {}
const Email: React.FC<Props> = () => {
  return <Image source={ICONS.email} resizeMode="contain" />;
};
const Phone: React.FC<Props> = () => {
  return <Image source={ICONS.phoneLightColor} resizeMode="contain" />;
};
const OpenEye: React.FC<Props> = () => {
  return <Image source={ICONS.openEye} resizeMode="contain" />;
};
const ClosedEye: React.FC<Props> = () => {
  return <Image source={ICONS.closedEye} resizeMode="contain" />;
};
export {Email, Phone, OpenEye, ClosedEye};
