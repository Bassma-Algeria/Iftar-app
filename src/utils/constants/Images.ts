const IMAGES = {
  latterns1: require('../../assets/images/latterns-1.png'),
  latterns2: require('../../assets/images/latterns-2.png'),
  noImage: require('../../assets/images/no-image.png'),
};

export {IMAGES};
