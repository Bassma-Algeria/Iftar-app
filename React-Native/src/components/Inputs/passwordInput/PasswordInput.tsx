import React from 'react';
import {
  TextInput,
  Pressable,
  Image,
  GestureResponderEvent,
  StyleProp,
  ViewStyle,
} from 'react-native';

import {styles} from '../BaseInput.style';

import {ICONS} from '../../../utils/constants/Icons';

import {BaseInput} from '../BaseInput';

interface Props {
  value: string;
  onTextChange: (text: string) => void;
  error?: string;
  placeholder: string;
  style?: StyleProp<ViewStyle>;
}

const PasswordInput: React.FC<Props> = props => {
  const [hidePassword, setHidePassword] = React.useState<boolean>(true);
  const togglePassword = () => setHidePassword(!hidePassword);

  return (
    <BaseInput {...props}>
      <EyeIcon hidePassword={hidePassword} togglePassword={togglePassword} />
      <TextInput
        style={styles.input}
        onChangeText={props.onTextChange}
        value={props.value}
        placeholder={props.placeholder}
        secureTextEntry={hidePassword}
      />
    </BaseInput>
  );
};

interface IconProps {
  hidePassword: boolean;
  togglePassword: (event: GestureResponderEvent) => void;
}

const EyeIcon: React.FC<IconProps> = ({hidePassword, togglePassword}) => {
  return (
    <Pressable onPress={togglePassword}>
      {hidePassword ? (
        <Image source={ICONS.closedEye} resizeMode="contain" />
      ) : (
        <Image source={ICONS.openEye} resizeMode="contain" />
      )}
    </Pressable>
  );
};

export {PasswordInput};
