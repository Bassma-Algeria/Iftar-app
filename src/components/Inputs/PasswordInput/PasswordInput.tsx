import React from 'react';
import {
  TextInput,
  Pressable,
  Image,
  GestureResponderEvent,
  StyleProp,
  ViewStyle,
} from 'react-native';

import {styles} from '../BaseInput.style';

import {ICONS} from '../../../utils/constants/Icons';

import {COLORS} from '../../../theme/Colors';

import {BaseInput} from '../BaseInput';

interface Props {
  value: string;
  onTextChange: (text: string) => void;
  error?: string;
  placeholder: string;
  style?: StyleProp<ViewStyle>;
}

const PasswordInput: React.FC<Props> = props => {
  const [hidePassword, setHidePassword] = React.useState<boolean>(true);
  const togglePassword = () => setHidePassword(!hidePassword);

  return (
    <BaseInput {...props}>
      <EyeIcon hidePassword={hidePassword} togglePassword={togglePassword} />
      <TextInput
        style={styles.input}
        onChangeText={props.onTextChange}
        value={props.value}
        placeholder={props.placeholder}
        secureTextEntry={hidePassword}
        placeholderTextColor={COLORS.grey}
      />
    </BaseInput>
  );
};

interface IconProps {
  hidePassword: boolean;
  togglePassword: (event: GestureResponderEvent) => void;
}

const EyeIcon: React.FC<IconProps> = ({hidePassword, togglePassword}) => {
  return (
    <Pressable onPress={togglePassword} testID="passwordIcon">
      <Image source={hidePassword ? ICONS.closedEye : ICONS.openEye} resizeMode="contain" />
    </Pressable>
  );
};

export {PasswordInput};
