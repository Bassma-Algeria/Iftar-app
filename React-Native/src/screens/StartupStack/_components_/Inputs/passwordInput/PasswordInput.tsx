import React from 'react';
import {TextInput, View, TouchableOpacity} from 'react-native';
import {styles} from './PasswordInput.style';
import {OpenEye, ClosedEye} from '../../Icons/Icons';
interface Props {}

const PasswordInput: React.FC<Props> = ({label}) => {
  const [input, onInputChange] = React.useState<string>('');
  const [hidePassword, setHidePassword] = React.useState<boolean>(true);
  const handleHidePasswordChange = () => {
    setHidePassword(!hidePassword);
  };
  return (
    <>
      <View style={styles.inputContainer}>
        <TouchableOpacity onPress={handleHidePasswordChange}>
          {hidePassword ? <ClosedEye /> : <OpenEye />}
        </TouchableOpacity>
        <TextInput
          style={styles.input}
          onChangeText={onInputChange}
          value={input}
          placeholder={label}
          secureTextEntry={hidePassword}
        />
      </View>
    </>
  );
};

export {PasswordInput};
