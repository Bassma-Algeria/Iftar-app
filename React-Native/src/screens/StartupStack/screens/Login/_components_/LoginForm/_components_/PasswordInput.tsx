import React from 'react';

import {styles} from '../../../Login.style';

import {useLoginContext} from '../_hooks_/UseLoginContext';

import {PasswordInput as BasePasswordInput} from '../../../../../../../components/Inputs/PasswordInput/PasswordInput';

const PasswordInput: React.FC = () => {
  const {loginInfo, setLoginInfo, setServerError} = useLoginContext();

  const onTextChange = (value: string) => {
    setServerError('');
    setLoginInfo({...loginInfo, password: value});
  };

  return (
    <BasePasswordInput
      value={loginInfo.password}
      onTextChange={onTextChange}
      style={styles.spaceInputs}
      placeholder="كلمة المرور"
    />
  );
};

export {PasswordInput};
