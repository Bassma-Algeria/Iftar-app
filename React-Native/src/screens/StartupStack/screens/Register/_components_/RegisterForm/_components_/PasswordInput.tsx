import React from 'react';

import {styles} from '../../../Register.style';

import {useRegisterContext} from '../_hooks_/UseRegisterContext';

import {PasswordInput as BasePasswordInput} from '../../../../../../../components/Inputs/PasswordInput/PasswordInput';

const PasswordInput: React.FC = () => {
  const {registerInfo, setRegisterInfo, setServerError} = useRegisterContext();

  const onTextChange = (value: string) => {
    setServerError('');
    setRegisterInfo({...registerInfo, password: value});
  };

  return (
    <BasePasswordInput
      value={registerInfo.password}
      onTextChange={onTextChange}
      style={styles.spaceInputs}
      placeholder="كلمة المرور"
    />
  );
};

export {PasswordInput};
