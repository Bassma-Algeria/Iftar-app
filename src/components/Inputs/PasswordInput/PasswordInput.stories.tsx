import React from 'react';

import {storiesOf} from '@storybook/react-native';

import {PasswordInput} from './PasswordInput';

const PasswordInputStories = storiesOf('PasswordInput', module);

const props: any = {placeholder: 'كلمة المرور'};

PasswordInputStories.add('PasswordInput', () => <PasswordInput {...props} />);
