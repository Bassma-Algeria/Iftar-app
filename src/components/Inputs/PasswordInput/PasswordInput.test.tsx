import {fireEvent, render} from '@testing-library/react-native';
import React from 'react';
import {Image} from 'react-native';
import renderer from 'react-test-renderer';

import {ICONS} from '../../../utils/constants/Icons';

import {PasswordInput} from './PasswordInput';

describe('PasswordInput Component', () => {
  const props: any = {placeholder: 'password'};

  it('should render correctly with the password hidden by default', () => {
    const tree = renderer.create(<PasswordInput {...props} />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should toggle the password when clicking the icon', () => {
    const instance = render(<PasswordInput {...props} />);

    const PasswordIconContainer = instance.getByTestId('passwordIcon');
    const Icon = PasswordIconContainer.findByType(Image);

    fireEvent(PasswordIconContainer, 'press');
    expect(Icon.props).toHaveProperty('source', ICONS.closedEye);

    fireEvent(PasswordIconContainer, 'press');
    expect(Icon.props).toHaveProperty('source', ICONS.openEye);
  });
});
