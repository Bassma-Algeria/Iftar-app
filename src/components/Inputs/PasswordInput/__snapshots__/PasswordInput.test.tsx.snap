// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PasswordInput Component should render correctly with the password hidden by default 1`] = `
<View
  pointerEvents="box-none"
>
  <View
    pointerEvents="none"
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 1,
        },
        undefined,
      ]
    }
  >
    <RNSVGSvgView
      bbHeight="100%"
      bbWidth={3}
      focusable={false}
      height="100%"
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          Object {
            "left": -2,
            "position": "absolute",
            "top": 10,
          },
          Object {
            "flex": 0,
            "height": "100%",
            "width": 3,
          },
        ]
      }
      width={3}
    >
      <RNSVGGroup>
        <RNSVGDefs>
          <RNSVGLinearGradient
            gradient={
              Array [
                0,
                335544320,
                1,
                0,
              ]
            }
            gradientTransform={null}
            gradientUnits={0}
            name="left"
            x1="1"
            x2="0"
            y1="0"
            y2="0"
          />
        </RNSVGDefs>
        <RNSVGRect
          fill={
            Array [
              1,
              "left",
            ]
          }
          height="100%"
          propList={
            Array [
              "fill",
            ]
          }
          width={2}
          x={0}
          y={-20}
        />
      </RNSVGGroup>
    </RNSVGSvgView>
    <RNSVGSvgView
      bbHeight="100%"
      bbWidth={3}
      focusable={false}
      height="100%"
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          Object {
            "left": "100%",
            "position": "absolute",
            "top": 10,
          },
          Object {
            "flex": 0,
            "height": "100%",
            "width": 3,
          },
        ]
      }
      width={3}
    >
      <RNSVGGroup>
        <RNSVGDefs>
          <RNSVGLinearGradient
            gradient={
              Array [
                0,
                335544320,
                1,
                0,
              ]
            }
            gradientTransform={null}
            gradientUnits={0}
            name="right"
            x1="0"
            x2="1"
            y1="0"
            y2="0"
          />
        </RNSVGDefs>
        <RNSVGRect
          fill={
            Array [
              1,
              "right",
            ]
          }
          height="100%"
          propList={
            Array [
              "fill",
            ]
          }
          width={2}
          x={0}
          y={-20}
        />
      </RNSVGGroup>
    </RNSVGSvgView>
    <RNSVGSvgView
      bbHeight={3}
      bbWidth="100%"
      focusable={false}
      height={3}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          Object {
            "left": 10,
            "position": "absolute",
            "top": "100%",
          },
          Object {
            "flex": 0,
            "height": 3,
            "width": "100%",
          },
        ]
      }
      width="100%"
    >
      <RNSVGGroup>
        <RNSVGDefs>
          <RNSVGLinearGradient
            gradient={
              Array [
                0,
                335544320,
                1,
                0,
              ]
            }
            gradientTransform={null}
            gradientUnits={0}
            name="bottom"
            x1="0"
            x2="0"
            y1="0"
            y2="1"
          />
        </RNSVGDefs>
        <RNSVGRect
          fill={
            Array [
              1,
              "bottom",
            ]
          }
          height={2}
          propList={
            Array [
              "fill",
            ]
          }
          width="100%"
          x={-20}
          y={0}
        />
      </RNSVGGroup>
    </RNSVGSvgView>
    <RNSVGSvgView
      bbHeight={3}
      bbWidth="100%"
      focusable={false}
      height={3}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          Object {
            "left": 10,
            "position": "absolute",
            "top": -2,
          },
          Object {
            "flex": 0,
            "height": 3,
            "width": "100%",
          },
        ]
      }
      width="100%"
    >
      <RNSVGGroup>
        <RNSVGDefs>
          <RNSVGLinearGradient
            gradient={
              Array [
                0,
                335544320,
                1,
                0,
              ]
            }
            gradientTransform={null}
            gradientUnits={0}
            name="top"
            x1="0"
            x2="0"
            y1="1"
            y2="0"
          />
        </RNSVGDefs>
        <RNSVGRect
          fill={
            Array [
              1,
              "top",
            ]
          }
          height={2}
          propList={
            Array [
              "fill",
            ]
          }
          width="100%"
          x={-20}
          y={0}
        />
      </RNSVGGroup>
    </RNSVGSvgView>
    <RNSVGSvgView
      bbHeight={13}
      bbWidth={13}
      focusable={false}
      height={13}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          Object {
            "left": -2,
            "position": "absolute",
            "top": -2,
          },
          Object {
            "flex": 0,
            "height": 13,
            "width": 13,
          },
        ]
      }
      width={13}
    >
      <RNSVGGroup>
        <RNSVGDefs>
          <RNSVGRadialGradient
            cx={12}
            cy={12}
            fx={12}
            fy={12}
            gradient={
              Array [
                0.8333333333333334,
                335544320,
                1,
                0,
              ]
            }
            gradientTransform={null}
            gradientUnits={1}
            name="topLeft"
            rx={12}
            ry={12}
          />
        </RNSVGDefs>
        <RNSVGPath
          d="M0,12 a12,12 0 0 1 12 -12 v2 v10 h-10 h-2 Z"
          fill={
            Array [
              1,
              "topLeft",
            ]
          }
          propList={
            Array [
              "fill",
            ]
          }
        />
      </RNSVGGroup>
    </RNSVGSvgView>
    <RNSVGSvgView
      bbHeight={13}
      bbWidth={13}
      focusable={false}
      height={13}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          Object {
            "left": "100%",
            "position": "absolute",
            "top": -2,
            "transform": Array [
              Object {
                "translateX": -10,
              },
            ],
          },
          Object {
            "flex": 0,
            "height": 13,
            "width": 13,
          },
        ]
      }
      width={13}
    >
      <RNSVGGroup
        matrix={
          Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ]
        }
      >
        <RNSVGDefs>
          <RNSVGRadialGradient
            cx={0}
            cy={12}
            fx={0}
            fy={12}
            gradient={
              Array [
                0.8333333333333334,
                335544320,
                1,
                0,
              ]
            }
            gradientTransform={null}
            gradientUnits={1}
            name="topRight"
            rx={12}
            ry={12}
          />
        </RNSVGDefs>
        <RNSVGPath
          d="M0,0 a12,12 0 0 1 12,12 h-2 h-10 v-10 v-2 Z"
          fill={
            Array [
              1,
              "topRight",
            ]
          }
          propList={
            Array [
              "fill",
            ]
          }
        />
      </RNSVGGroup>
    </RNSVGSvgView>
    <RNSVGSvgView
      bbHeight={13}
      bbWidth={13}
      focusable={false}
      height={13}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          Object {
            "left": -2,
            "position": "absolute",
            "top": "100%",
            "transform": Array [
              Object {
                "translateY": -10,
              },
            ],
          },
          Object {
            "flex": 0,
            "height": 13,
            "width": 13,
          },
        ]
      }
      width={13}
    >
      <RNSVGGroup
        matrix={
          Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ]
        }
      >
        <RNSVGDefs>
          <RNSVGRadialGradient
            cx={12}
            cy={0}
            fx={12}
            fy={0}
            gradient={
              Array [
                0.8333333333333334,
                335544320,
                1,
                0,
              ]
            }
            gradientTransform={null}
            gradientUnits={1}
            name="bottomLeft"
            rx={12}
            ry={12}
          />
        </RNSVGDefs>
        <RNSVGPath
          d="M12,12 a12,12 0 0 1 -12,-12 h2 h10 v10 v2 Z"
          fill={
            Array [
              1,
              "bottomLeft",
            ]
          }
          propList={
            Array [
              "fill",
            ]
          }
        />
      </RNSVGGroup>
    </RNSVGSvgView>
    <RNSVGSvgView
      bbHeight={13}
      bbWidth={13}
      focusable={false}
      height={13}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          Object {
            "left": "100%",
            "position": "absolute",
            "top": "100%",
            "transform": Array [
              Object {
                "translateX": -10,
              },
              Object {
                "translateY": -10,
              },
            ],
          },
          Object {
            "flex": 0,
            "height": 13,
            "width": 13,
          },
        ]
      }
      width={13}
    >
      <RNSVGGroup
        matrix={
          Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ]
        }
      >
        <RNSVGDefs>
          <RNSVGRadialGradient
            cx={0}
            cy={0}
            fx={0}
            fy={0}
            gradient={
              Array [
                0.8333333333333334,
                335544320,
                1,
                0,
              ]
            }
            gradientTransform={null}
            gradientUnits={1}
            name="bottomRight"
            rx={12}
            ry={12}
          />
        </RNSVGDefs>
        <RNSVGPath
          d="M12,0 a12,12 0 0 1 -12,12 v-2 v-10 h10 h2 Z"
          fill={
            Array [
              1,
              "bottomRight",
            ]
          }
          propList={
            Array [
              "fill",
            ]
          }
        />
      </RNSVGGroup>
    </RNSVGSvgView>
    <RNSVGSvgView
      bbHeight="100%"
      bbWidth="100%"
      focusable={false}
      height="100%"
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          Object {
            "position": "absolute",
          },
          Object {
            "flex": 0,
            "height": "100%",
            "width": "100%",
          },
        ]
      }
      width="100%"
    >
      <RNSVGGroup>
        <RNSVGDefs>
          <RNSVGMask
            height="100%"
            maskContentUnits={1}
            maskTransform={
              Array [
                1,
                0,
                0,
                1,
                0,
                0,
              ]
            }
            maskUnits={0}
            name="maskPaintBelow"
            width="100%"
            x="0%"
            y="0%"
          >
            <RNSVGRect
              fill={4294967295}
              height="100%"
              propList={
                Array [
                  "fill",
                ]
              }
              width="100%"
              x={0}
              y={0}
            />
            <RNSVGRect
              fill={4278190080}
              height={10}
              propList={
                Array [
                  "fill",
                ]
              }
              width={10}
              x={0}
              y={0}
            />
            <RNSVGRect
              fill={4278190080}
              height={10}
              matrix={
                Array [
                  1,
                  0,
                  0,
                  1,
                  -10,
                  0,
                ]
              }
              propList={
                Array [
                  "fill",
                ]
              }
              width={10}
              x="100%"
              y={0}
            />
            <RNSVGRect
              fill={4278190080}
              height={10}
              matrix={
                Array [
                  1,
                  0,
                  0,
                  1,
                  0,
                  -10,
                ]
              }
              propList={
                Array [
                  "fill",
                ]
              }
              width={10}
              x={0}
              y="100%"
            />
            <RNSVGRect
              fill={4278190080}
              height={10}
              matrix={
                Array [
                  1,
                  0,
                  0,
                  1,
                  -10,
                  -10,
                ]
              }
              propList={
                Array [
                  "fill",
                ]
              }
              width={10}
              x="100%"
              y="100%"
            />
          </RNSVGMask>
        </RNSVGDefs>
        <RNSVGRect
          fill={4278190080}
          fillOpacity={0.08}
          height="100%"
          mask="maskPaintBelow"
          propList={
            Array [
              "fill",
              "fillOpacity",
            ]
          }
          width="100%"
          x={0}
          y={0}
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
  <View
    onLayout={[Function]}
    pointerEvents="box-none"
    style={
      Array [
        Object {
          "alignSelf": "flex-start",
        },
        undefined,
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "white",
            "borderRadius": 10,
            "flexDirection": "row",
            "height": 60,
            "justifyContent": "space-between",
            "paddingLeft": 18,
            "paddingRight": 20,
            "width": "100%",
          },
          Object {},
        ]
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        testID="passwordIcon"
      >
        <Image
          resizeMode="contain"
          source={
            Object {
              "process": [Function],
            }
          }
        />
      </View>
      <TextInput
        placeholder="password"
        placeholderTextColor="#C4C4C4"
        secureTextEntry={true}
        style={
          Object {
            "color": "#000",
            "flex": 1,
            "fontFamily": "Cairo-Regular",
            "textAlign": "right",
          }
        }
      />
    </View>
  </View>
</View>
`;
